sbc_compassion/migrations/10.0.1.6.0/post-migration.py

```diff
@@ -0,0 +1,196 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2019 Compassion CH (http://www.compassion.ch)
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+import logging
+import base64
+from io import BytesIO
+from openupgradelib import openupgrade
+
+_logger = logging.getLogger(__name__)
+
+try:
+    import PyPDF2
+    from PIL import Image
+
+    from pdfminer.converter import PDFPageAggregator
+    from pdfminer.layout import LAParams, LTTextBox, LTTextLine
+    from pdfminer.pdfdocument import PDFDocument
+    from pdfminer.pdfinterp import PDFPageInterpreter, PDFResourceManager
+    from pdfminer.pdfpage import PDFPage
+    from pdfminer.pdfparser import PDFParser
+
+except ImportError:
+    _logger.error("Python libraries 'PyPDF2', 'pdfminer' and 'Pillow' are required for"
+                  " the correspondences migration in order to extract images from PDFs")
+
+
+@openupgrade.migrate(use_env=True)
+def migrate(env, version):
+    """
+    move the `original_text` to `english_text`
+    then, try to extract texts and images from the PDF. They will be stored in
+    `original_attachment_ids` and `original_text`
+    """
+    if not version:
+        return
+
+    # Filters correspondences that can safely drop the PDF and regenerate it later
+    correspondences = env['correspondence'].search([
+        '&',
+        '&',
+        '&',
+        '&',
+        '|',
+        ('source', '=', 'website'),
+        ('source', '=', 'compassion'),
+        ('letter_format', '=', 'pdf'),
+        ('direction', '=', 'Supporter To Beneficiary'),
+        ('template_id', '!=', False),
+        ('store_letter_image', '=', True)
+    ])
+
+    templates_backgrounds_sizes = retrieve_background_sizes(env)
+
+    correspondences = correspondences.filtered(lambda x: x.id in [27788, 108590])
```

Comment:
I think this was for your testing purposes. However I suggest that we split the recordset into small batches (let's say 10 correspondences) and launch the migration of these batches into job queues functions, which are asynchronous.

This way, the migration won't block the launching of the server and it can be done gradually while we can still use Odoo.